backend/functions/views.py

```diff
@@ -111,37 +261,185 @@ def copy_from_ocr_document_to_block_text(request):
-                parent_data = ocr_document,
+                parent_data=ocr_document,
                 language=ocr_document.language,
                 text=text,
                 domain=ocr_document.ocr_domain,
-                instance_id = export_dataset_instance,
+                instance_id=export_dataset_instance,
             )
             all_block_texts.append(block_text)
-            task.metadata_json["copy_from_ocr_document_to_block_text"]=True
+            task.metadata_json["copy_from_ocr_document_to_block_text"] = True
             task.task_status = FREEZED
             task.save()
 
-
     # TODO: implement bulk create if possible (only if non-hacky)
     for block_text in all_block_texts:
         block_text.save()
     # dataset_models.SentenceText.objects.bulk_create(all_sentence_texts)
-    
+
     # project.metadata_json["copy_from_ocr_document_to_block_text"]=True
     # project.save()
-    ret_dict = {"message": "SUCCESS!"}         
+    ret_dict = {"message": "SUCCESS!"}
+    ret_status = status.HTTP_200_OK
+    return Response(ret_dict, status=ret_status)
+
+
+@api_view(["POST"])
+def schedule_google_translate_job(request):
+    """
+    Schedules a Google Translate job for a given dataset instance
+
+    Request Body
+    {
+        "input_dataset_instance_id": <int>,
+        "languages": <list>
+        "output_dataset_instance_id": <int>
+    }
+
+    Response Body
+    {
+        "message": <str>
+        "result": <str>
+        "status": DjangoStatusCode
+    }
+    """
+
+    # Get the post request data
+    input_dataset_instance_id = request.data["input_dataset_instance_id"]
+    languages = request.data["languages"]
+    output_dataset_instance_id = request.data["output_dataset_instance_id"]
+
+    # Convert string list to a list
+    languages = ast.literal_eval(languages)
+
+    # Perform checks on the input and output dataset instances
+    dataset_instance_check_status = check_translation_function_inputs(
+        input_dataset_instance_id, output_dataset_instance_id
+    )
+
+    if dataset_instance_check_status["status"] in [
+        status.HTTP_400_BAD_REQUEST,
+        status.HTTP_404_NOT_FOUND,
+    ]:
+        return Response(
+            {"message": dataset_instance_check_status["message"]},
+            status=status.HTTP_400_BAD_REQUEST,
+        )
+
+    # Collect the sentences from Sentence Text based on dataset id
+    input_sentences = list(
+        dataset_models.SentenceText.objects.filter(
+            instance_id=input_dataset_instance_id
+        ).values_list(
+            "id",
+            "corrected_text",
+            "language",
+            "context",
+            "quality_status",
+            "metadata_json",
+        )
+    )
+
+    # Call the function to save the TranslationPair dataset
+    save_status = save_translation_pairs(
+        languages,
+        input_sentences,
+        output_dataset_instance_id,
+        batch_size=128,
+        api_type="google",
+    )
```

Comment:
I suppose this would be a very time-consuming API call even for a dataset instance with 10k sentences. Shall we make this an async API?

Sorry I should have noticed this earlier. But I had mentioned in [SN-245](https://project-sunbird.atlassian.net/browse/SN-245) description that it should be a Celery task.

Can you please implement it that way?

Reply:
Yeah, I haven't done that currently because that might involve a lot of breaking changes outside the `functions` module. Can we make a separate ticket for converting this to a celery task after this is merged to dev?

Reply:
OK sure, makes sense. Can you maybe try the custom `bulk_create` thing (mentioned above) ?
Once that's done, you can request Ishvinder to test it out and merge.

You can also create a ticket yourself on Jira and assign to yourself for the Celery changes.

Reply:
Please also check if auto-increment works (the prev answer to that answer has mentioned it does not work)

Reply:
Sure, sounds good. I will try out the bulk create thing.